Sources/MapboxDirections/DirectionsOptions.swift

```diff
@@ -129,6 +129,44 @@ open class DirectionsOptions: Codable {
         self.profileIdentifier = profileIdentifier ?? .automobile
     }
     
+    /**
+     Creates new options object by deserializing given `url`
+     
+     Initialization fails if it is unable to extract `waypoints` list and `profileIdentifier`. If other properties are failed to decode - it will just skip them.
+     
+     - parameter url: An URL, used to make a route request.
+     */
+    public convenience init?(url: URL) {
```

Comment:
What’s the difference between `init(url:)` and `restore(from:)`? Why would the developer call `restore(from:)` on an existing RouteOptions object instead of initializing a new RouteOptions object? Sorry that https://github.com/mapbox/mapbox-directions-swift/pull/655#discussion_r810275117 didn’t make this point clearly.

Reply:
Well, my original thought was that this method would unlock use-cases where user would want "the same exact type of route, but between different points". But now I see that it is simply achieved just by having an `init`. Probably should make `restore(from:)` internal to avoid cluttering the API.

Reply:
It looks like the distinction is actually between URL path components and URL query items. How about folding everything into the initializer, then factoring out an initializer that takes the waypoints, profile identifier, and an array of query items?

Reply:
Updated. Init now looks a bit more than I would love to, and I am not 100% sure about that mix of arguments on the `DirectionsOptions(waypoints:profileIdentifier:queryItems:)` as it has some SDK types together with url request parts in the same row which might look weird.  Other than that, API looks more natural to me.